controller/api/destination/server.go

```diff
@@ -225,7 +226,12 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	// up to the fallbackProfileListener to merge updates from the primary and
 	// secondary listeners and send the appropriate updates to the stream.
 	if dest.GetContextToken() != "" {
-		profile, err := profileID(path, dest.GetContextToken(), s.clusterDomain)
+		ctxToken, err := parseContextToken(dest.GetContextToken())
+		if err != nil {
+			log.Errorf("error parsing context token %s as json:%s", dest.GetContextToken(), err)
```

Comment:
Ah, I see: you're returning the error in order to log that we have received an old format token.  I think this is confusing because the `parseContextToken` method is actually successful, so I don't think it should return an error in this case.  Instead, I would have the `parseContextToken` method itself log when it gets a token in the old format.  I'd also probably just make that log at the warning level since getting old format tokens doesn't cause any problems (but users will want to upgrade to the new format eventually).

Reply:
@adleong this is a fair point, I've read the previous comments as well. I have made some changes to log both when the JSON/context token is invalid (as an error) and when a warning when the format of the token is old. I wanted to avoid checking if the JSON is valid since it's a pretty expensive operation, but that's also an alternative to make it more cleaner if the current way is not very straightforward. I'm thinking the alternative would be:
```go
func parseContextToken(token string) {
  ctxToken := &contextToken
  if !isValidJSON(token) {
      // process old format
   }

   // unamarshal

  return ctxToken
}
```